src/dbnode/server/server.go

```diff
@@ -1223,7 +1222,11 @@ func withEncodingAndPoolingOptions(
 		SetCheckedBytesPool(bytesPool).
 		SetResultsPool(resultsPool)
 
-	resultsPool.Init(func() index.Results { return index.NewResults(indexOpts) })
+	resultsPool.Init(func() index.Results {
+		// NB(r): Need to initialize after setting the index opts so
+		// it seems the same reference of the options as is set
```

Comment:
This sentence reads like word salad lol. I think you were trying to say: "so it sees the same reference to the indexOptions"